### ISSUE TITLE
License placement policy in modules

### ISSUE THREAD

**Issue by elicoidal** [OCA/maintainer-tools#86]
There is no mention in the contributing.md file about how to insert the license in the files.
AFAIK the latest policy is to have a LICENSE.txt and long descriptor in the __openerp__.py and make reference of it in the other py files.

I think we should add it as a guideline

**Comment by dreispt** [comment]
This is a good to excuse to discuss how licenses should be described in the py files.
@odony [said](https://github.com/odoo/odoo/commit/0bd454534818941bc3216025ebe3dcac3d78c19d) that "explicit boilerplate should not be necessary - copyright law applies automatically".

Personally I like the idea of simplifying license boilerplate at the top of code files, but I think that you should still be explicit of what is the license and authors, because in case the file is taken apart from the module, you can still have that basic information.

**Comment by elicoidal** [comment]
I like the idea of large boiler plate in openerp.py + simple one on other files + add license logo.
Shall we set it as a guideline even if different from Odoo?
I tend to stick to Odoo's guidelines but we have already some other different rules (eg: PEP8) so I would be OK to diverge here

**Comment by dreispt** [comment]
I'm seeing people totally removing copyright and license information from code files, replacing it with a "see the LICENSE file" or the "__openerp__.py file".

IMO there are good reasons to not do so, but the fact is that there is nothing in the contributing guidelines preventing it.

For reference, this has also been discussed here: https://github.com/OCA/maintainer-tools/pull/91

**Comment by Yajo** [comment]
> I like the idea of large boiler plate in openerp.py + simple one on other files

IMHO that does not make much sense. Individual headers are for preventing that a file is picked to another project with a different license and loses the AGPL accidentally, which applies equally to `__openerp__.py` and other `.py` files.

Well, not equally... In fact, `__openerp__.py` does not contain code at all. It could be an XML or JSON file. Yes, it ends with `.py`, but it really is a data file.

**Comment by simahawk** [comment]
Just spotted a new comer's PR adding a new (C) owner just for migration, which usually we don't do.
Still, I struggled to find official guidelines on this topic here https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md
It's me, or we really miss few lines about this?

**Comment by pedrobaeza** [comment]
Yeah, there's no official statement about this. Sometimes it's difficult to define the border.

**Comment by simahawk** [comment]
Well, can we simply state that if we don't change at least, say, 3 lines (or even 1?) of code we are not allowed to add your company/yourself to (C)?
Still, it's nice to add the guy to contribs.

**Comment by leemannd** [comment]
IMO we could state that copyright comes with creative work.
You should get rewarded by adding yourself to the contibutors. Copyright needs at least adding features, bugfixes, rewriting the module.
@simahawk By any chance can you point us to some example?

**Comment by dreispt** [comment]
AFAIK migrating code (even porting to other programming language) does not grant copyright, similarly to what generally happens with book translations (original author keeps the copyright, translator does not acquire copyrights).
